planner/core/rule_partition_processor.go

```diff
@@ -110,12 +110,33 @@ func (s *partitionProcessor) prune(ds *DataSource) (LogicalPlan, error) {
 	}
 	partitionDefs := ds.table.Meta().Partition.Definitions
 
+	sctx := ds.SCtx()
+	filterConds := make([]expression.Expression, len(ds.allConds))
+	copy(filterConds, ds.allConds)
+	filterConds = expression.PropagateConstant(sctx, filterConds)
+	presolvedFilters := solver.Solve(sctx, filterConds)
+	alwaysFalse := false
+	if len(presolvedFilters) == 1 {
+		// Constant false.
+		if con, ok := presolvedFilters[0].(*expression.Constant); ok && con.DeferredExpr == nil && con.ParamMarker == nil {
+			ret, _, err := expression.EvalBool(sctx, expression.CNFExprs{con}, chunk.Row{})
+			if err == nil && ret == false {
+				alwaysFalse = true
+			}
+		}
+	}
+	if alwaysFalse {
```

Comment:
I don't understand this optimization, when could the `alwaysFalse` condition hold, Can you give me an example?